### DISCUSSION TITLE
Collection name constraints

### DISCUSSION THREAD

**Discussion by basavarajdodamani** [strapi/strapi#7933] [Legacy Topics]
Hello, could anyone confirm if we have constraints with collection naming.  A query written on single word collection doesnt work fine when same query applied on other collection having two words in its name.
**Below works:**
      const api = await strapi.services.enterprises.find({enterprise_id:id});
**Doesnt work:**
      const api = await strapi.services.api-details.find({enterprise_id:id});

Error:
![image](https://user-images.githubusercontent.com/69140968/93463738-cb594780-f905-11ea-862a-8a2492aa437f.png)

Any hints to resolve this issue or should i change my collection name to single word **apidetails**?

Kind Regards,
Basu

**Comment by MattieBelt** [comment]
Hey @basavarajdodamani,

If an JS object property name contains a _-_ you can't use the _Dot notation_, instead you can use the _Bracket notation_ ([source](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Property_accessors)):
```js
const api = await strapi.services['api-details'].find({enterprise_id:id});
```
The above should work.

**Reply by basavarajdodamani** [reply]
@MattieBelt Thanks a lot for your quickest response. It works fine!
